packages/react-core/src/components/DualListSelector/DualListSelectorPane.tsx

```diff
@@ -0,0 +1,202 @@
+  focusedOption: string;
+}
+
+export class DualListSelectorPane extends React.Component<DualListSelectorPaneProps, DualListSelectorPaneState> {
+  static displayName = 'DualListSelectorPane';
+  static defaultProps: PickOptional<DualListSelectorPaneProps> = {
+    isChosen: false,
+    status: '',
+    title: '',
+    options: [] as React.ReactNode[],
+    selectedOptions: [],
+    isSearchable: false
+  };
+  private menuEl = React.createRef<HTMLDivElement>();
+  private optionsRefs = [] as HTMLElement[];
+
+  constructor(props: DualListSelectorPaneProps) {
+    super(props);
+    this.state = {
+      input: '',
+      focusedOption: null
+    };
+  }
+
+  onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ input: e.target.value });
+  };
+
+  sendRef = (optionRef: React.ReactNode, index: number) => {
+    this.optionsRefs[index] = optionRef as HTMLElement;
+  };
+
+  handleKeys = (event: KeyboardEvent) => {
+    const key = event.key;
+    let moveFocus = false;
+    let currentIndex = -1;
+    if (['ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'].includes(event.key)) {
+      if (document.activeElement === this.menuEl.current) {
+        currentIndex = 0;
+        moveFocus = true;
+      } else {
+        this.optionsRefs.forEach((option, index) => {
+          if (document.activeElement === option) {
+            currentIndex = key === 'ArrowUp' || key === 'ArrowLeft' ? index - 1 : index + 1;
+            moveFocus = true;
+          }
+        });
+      }
+      event.preventDefault();
+    }
+    if (moveFocus && this.optionsRefs[currentIndex]) {
+      this.optionsRefs[currentIndex].focus();
+      this.setState({ focusedOption: `${this.props.id}-option-${currentIndex}` });
+    }
+  };
+
+  displayOption = (option: React.ReactNode, input: string) => {
+    if (this.props.filterOption) {
+      return this.props.filterOption(option, input);
+    } else {
+      return option
+        .toString()
+        .toLowerCase()
+        .includes(input.toLowerCase());
+    }
+  };
+
+  onOptionSelect = (e: React.MouseEvent | React.ChangeEvent, index: number, isChosen: boolean) => {
+    this.setState({ focusedOption: `${this.props.id}-option-${index}` });
+    this.props.onOptionSelect(e, index, isChosen);
+  };
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.handleKeys);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.handleKeys);
+  }
+
+  render() {
+    const {
+      isChosen,
+      title,
+      actions,
+      isSearchable,
+      className,
+      status,
+      selectedOptions,
+      options,
+      id,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      filterOption,
+      onOptionSelect,
+      ...props
+    } = this.props;
+    const { input, focusedOption } = this.state;
+
+    return (
+      <div
+        className={css(styles.dualListSelectorPane, isChosen ? styles.modifiers.chosen : 'pf-m-available', className)}
```

Comment:
`styles.modifiers.available`? Or placeholder css / not on the styles object?

Reply:
exactly, @kmcfaul

Reply:
@mcoker should these style `pf-m-available` be added to the component?